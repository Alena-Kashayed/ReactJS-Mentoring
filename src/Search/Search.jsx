import React from 'react';
import styles from './Search.scss';
import SearchTypeSwitcher from './SearchTypeSwitcher/SearchTypeSwitcher';

const Search = () => (
  <section className={styles.searchWrapper}>
    <div className={styles.search}>
      <div className={styles.content}>
        <div className={styles.logo}>Netflixtrullete</div>
        <label className={styles.searchLabel} htmlFor="search">
          Find your movie
        </label>
        <input
          className={styles.searchInput}
          type="text"
          placeholder="Quentin Tarantino"
        />
        <SearchTypeSwitcher />
      </div>
    </div>
  </section>
);

export default Search;
