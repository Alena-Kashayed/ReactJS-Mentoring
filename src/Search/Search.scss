.search {
  border: 1px solid #DDDDDD;
  padding: 15px 60px 20px;
  width: 100%;
  color: #F55263;
  background-image: url("../assets/img/popsignuptwoweeks-perspective_alpha_website_small.jpg");
  background-repeat: no-repeat;
  background-position: 50% 0;
  font-weight: 600;
  display: flex;
  position: relative;
  box-sizing: border-box;
  &::before {
    content: "";
    display: block;
    height: 100%;
    width: 100%;
    position: absolute;
    top: 0;
    left: 0;
    background-color: rgba(0, 0, 0, 0.5);
  }
  .content {
    z-index: 2;
    display: flex;
    flex-direction: column;
    width: 100%;

    .logo {
      margin-bottom: 30px;
    }
    .searchLabel {
      color: #FFFFFF;
      text-transform: uppercase;
      margin-bottom: 15px;
    }
    .searchInput {
      background-color: #000000;
      padding: 10px 15px;
      width: 100%;
      border: none;
      border-bottom: 2px solid #F55263;
      margin-bottom: 10px;
      color: #FFFFFF;;
      /////  placeholder styles
      &:focus::-webkit-input-placeholder {color: transparent}
      &:focus::-moz-placeholder          {color: transparent}
      &:focus:-moz-placeholder           {color: transparent}
      &:focus:-ms-input-placeholder      {color: transparent}
    }
  }
}
