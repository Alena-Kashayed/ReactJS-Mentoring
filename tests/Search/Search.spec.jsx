import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import Search from '../../src/Search/Search';

const mockStore = configureStore();

describe('Search', () => {
  it('is rendered correctly', () => {
    const wrapper = shallow(<Search
      store={mockStore({})}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
