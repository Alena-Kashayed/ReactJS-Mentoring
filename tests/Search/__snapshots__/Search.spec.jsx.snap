// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search is rendered correctly 1`] = `
<Search
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
